collectors/channels_collector.go

```diff
@@ -317,6 +340,163 @@ func (c *ChannelsCollector) Collect(ch chan<- prometheus.Metric) {
 			)
 		}
 	}
+
+	// Get all remote policies
+	remotePolicies, err := c.getRemotePolicies(getInfoResp.IdentityPubkey)
+	if err != nil {
+		channelLogger.Error(err)
+		return
+	}
+
+	// Export the inbound fee metric for a series of amounts.
+	var receiveAmt btcutil.Amount = 100000
+	for {
+		// For each fee amount, we'll approximate the total routing fee
+		// that needs to be paid to pay us.
+		inboundFee := approximateInboundFee(
+			receiveAmt, remotePolicies, remoteBalances,
+		)
+		if inboundFee == nil {
+			break
+		}
+
+		// Calculate the fee proportional to the amount to receive.
+		proportionalFee := float64(*inboundFee) / float64(receiveAmt)
+
+		ch <- prometheus.MustNewConstMetric(
+			c.inboundFee, prometheus.GaugeValue,
+			proportionalFee,
+			receiveAmt.String(),
+		)
+
+		// Continue the series with double the amount.
+		receiveAmt *= 2
+	}
+}
+
+// approximateInboundFee calculates to forward fee for a specific amount charged by the
+// last hop before this node.
+func approximateInboundFee(amt btcutil.Amount, remotePolicies map[uint64]*lnrpc.RoutingPolicy,
+	remoteBalances map[uint64]btcutil.Amount) *btcutil.Amount {
+
+	var fee btcutil.Amount
+
+	// Copy the remote balances so they can be decreased as we find shards.
+	remainingBalances := make(map[uint64]btcutil.Amount)
+	for ch, balance := range remoteBalances {
+		remainingBalances[ch] = balance
+	}
+
+	// Assume a perfect mpp splitting algorithm that knows exactly how much
+	// can be sent through each channel. This is a simplification, because
+	// in reality senders need to trial and error to find a shard amount
+	// that works.
+	//
+	// We'll keep iterating through all channels until we've covered the
+	// total amount. Each iteration, the best channel for that shard is
+	// selected based on the specific fee.
+	amountRemaining := amt
+	for amountRemaining > 0 {
```

Comment:
Is this an accurate reflection of our available inbound if we don't limit the number of shards? Maybe I'm missing it in here, but it seems like we should be taking that into account, because they can't actually reach us if we exceed shard max.

Reply:
Shard max is configurable, so we can't make assumptions on that.

Reply:
lnd's default is probably a fair representation of reality, but we can always come back and refine.